e2e/framework/utils/k8s_helper.go

```diff
@@ -251,3 +251,60 @@ func (k8sh *K8sHelper) IsThirdPartyResourcePresent(tprname string) bool {
 
 	return false
 }
+
+func (k8sh *K8sHelper) GetPodDetails(podNamePattern string, namespace string) (string, error) {
+	cmdArgs := []string{"get", "pods", "-l", "app=" + podNamePattern, "-o", "wide"}
+	if namespace != "" {
+		cmdArgs = append(cmdArgs, []string{"-n", namespace}...)
+	}
+	sout, serr, status := ExecuteCmd("kubectl", cmdArgs)
+	if status != 0 {
+		return serr, fmt.Errorf("Cannot find service")
+	}
+	return sout, nil
+}
+
+func (k8sh *K8sHelper) GetPodHostId(podNamePattern string, namespace string) (string, error) {
+	data, err1 := k8sh.GetPodDetails(podNamePattern, namespace)
+	if err1 != nil {
+		return data, err1
+	}
+
+	// Handle case when no data is returned
+	lines := strings.Split(data, "\n")
+	if len(lines) <= 1 {
+		return "NOT FOUND", fmt.Errorf("Cannot find Pod with Name matching %s", podNamePattern)
+	}
+
+	// Handle case when no data is returned
+	lines = lines[1 : len(lines)-1]
+	if len(lines) < 1 {
+		return "NOT FOUND", fmt.Errorf("Cannot find Pod with Name matching %s", podNamePattern)
+	}
+
+	//extract name of the pod
+	lineRawdata := strings.Split(lines[0], "  ")
+	var r []string
+	for _, str := range lineRawdata {
+		if str != "" {
+			r = append(r, strings.TrimSpace(str))
+		}
+	}
+
+	//get host Ip of the pod
+	cmdArgs := []string{"get", "pods", r[0], "-o", "json"}
```

Comment:
this may be easier using output type `jsonpath` instead.  Essentially you pass a query path to use on the returned JSON for the field you care about, and then you could skip the deserialization and querying you are doing below.